django/db/models/deletion.py

```diff
@@ -258,8 +261,10 @@ def sort(self):
                                 for model in sorted_models)
 
     def delete(self):
+        all_instances = []
         # sort instance collections
         for model, instances in self.data.items():
+            all_instances.extend(instances)
```

Comment:
Not sure it's worth creating a possibly large list for the sole purpose of fetching the first object. Maybe `deque([], 2)` could be used of the following logic used below

```python
if len(self.data) == 1 and len(instances) == 1 and self.can_fast_delete(instances[0]):
    with transaction.mark_for_rollback_on_error():
        count = sql.DeleteQuery(model).delete_batch([instances[0].pk], self.using)
    return count, {model._meta.label: count}
```

That's possible because `len(self.data) == 1` implies `model` and `instances` were necessarily assigned here.